monai/fl/client/monai_algo.py

```diff
@@ -38,77 +48,170 @@ def convert_global_weights(global_weights, local_var_dict):
+
+        Args:
+            extra: Dict with additional information that should be provided by FL system,
+            i.e., `ExtraItems.CLIENT_NAME` and `ExtraItems.APP_ROOT`.
+
+        """
         if extra is None:
             extra = {}
         self.logger.info(f"Initializing {self.client_name} ...")
         self.client_name = extra.get(ExtraItems.CLIENT_NAME, "noname")
 
+        # FL platform needs to provide filepath to configuration files
+        self.app_root = extra.get(ExtraItems.APP_ROOT, "")
+
         # Read bundle config files
-        config_files = []
-        if self.config_train_file:
-            config_files.append(self.config_train_file)
-        if self.config_evaluate_file:
-            config_files.append(self.config_evaluate_file)
-        if self.config_filters_file:
-            config_files.append(self.config_filters_file)
+        self.bundle_root = os.path.join(self.app_root, self.bundle_root)
+
+        config_train_files = []
+        config_eval_files = []
+        config_filter_files = []
+        if self.config_train_filename:
+            config_train_files.append(os.path.join(self.bundle_root, self.config_train_filename))
+            config_eval_files.append(os.path.join(self.bundle_root, self.config_train_filename))
+        if self.config_evaluate_filename:
+            config_eval_files.append(os.path.join(self.bundle_root, self.config_evaluate_filename))
+        if self.config_filters_filename:  # filters are read by train_parser
+            config_filter_files.append(os.path.join(self.bundle_root, self.config_filters_filename))
 
         # Parse
-        self.parser = ConfigParser()
-        self.parser.read_config(config_files)
-        self.parser.parse()
-
-        # Get trainer, evaluator, and filters
-        self.trainer = self.parser.get_parsed_content(
-            BundleConst.KEY_TRAINER, default=ConfigItem(None, BundleConst.KEY_TRAINER)
+        self.train_parser = ConfigParser()
+        self.eval_parser = ConfigParser()
+        self.filter_parser = ConfigParser()
+        if len(config_train_files) > 0:
+            self.train_parser.read_config(config_train_files)
+            check_bundle_config(self.train_parser)
+        if len(config_eval_files) > 0:
+            self.eval_parser.read_config(config_eval_files)
+            check_bundle_config(self.eval_parser)
+        if len(config_filter_files) > 0:
+            self.filter_parser.read_config(config_eval_files)
+
+        # override some config items
+        self.train_parser[RequiredBundleKeys.BUNDLE_ROOT] = self.bundle_root
+        self.eval_parser[RequiredBundleKeys.BUNDLE_ROOT] = self.bundle_root
+        # number of training epochs for each round
+        if BundleKeys.TRAIN_TRAINER_MAX_EPOCHS in self.train_parser:
+            self.train_parser[BundleKeys.TRAIN_TRAINER_MAX_EPOCHS] = self.local_epochs
+
+        self.train_parser.parse()
+        self.eval_parser.parse()
```

Comment:
I think no need to call the `parse()` explicitly, it can "lazy-parse" when calling `get_parsed_content()`.
Anyway, it's not a problem.

Thanks.

Reply:
ok. Good to know.